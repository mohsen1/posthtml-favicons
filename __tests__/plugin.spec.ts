import posthtml from "posthtml";

import plugin from "../src";
import { Configuration, FavIconResponse } from "favicons";

jest.mock("fs");
jest.mock("favicons", () => (input: string, config: any, cb: Function) => {
    setTimeout(() => {
        const res: FavIconResponse = {
            images: [{ name: "foo", contents: Buffer.alloc(10) }],
            files: [{ name: "bar", contents: Buffer.alloc(10) }],
            html: ['<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">']
        };
        cb(null, res);
    }, 1);
});

describe("basics", () => {
    it("does nothing if a link tag does not exist", async () => {
        const html = `
            <!doctype html>
            <html>
                <head>
                </head>
                <body>
                </body>
            </html>
        `;
        const config: Partial<Configuration> = {
            icons: {
                android: false,
                appleIcon: false,
                appleStartup: false,
                coast: false,
                favicons: true,
                firefox: false,
                yandex: false,
                windows: false
            }
        };
        const result = await posthtml()
            .use(plugin({ configuration: config }))
            .process(html);
        expect(result.html).toMatchSnapshot();
    });

    it("appends basic tags", async function() {
        const iconPath = "./__tests__/icon.png";
        const html = `
            <!doctype html>
            <html>
                <head>
                    <link rel="icon" href="${iconPath}" />
                </head>
                <body>
                </body>
            </html>
        `;
        const config: Partial<Configuration> = {
            icons: {
                android: false,
                appleIcon: false,
                appleStartup: false,
                coast: false,
                favicons: true,
                firefox: false,
                yandex: false,
                windows: false
            }
        };

        const result = await posthtml()
            .use(plugin({ configuration: config }))
            .process(html);
        expect(result.html).toMatchSnapshot();
    });
});
