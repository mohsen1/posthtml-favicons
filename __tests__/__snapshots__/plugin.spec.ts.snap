// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basics appends basic tags 1`] = `
"
            <!doctype html>
            <html>
                <head>
                    
                <link rel=\\"icon\\" type=\\"image/png\\" sizes=\\"16x16\\" href=\\"/favicon-16x16.png\\"></head>
                <body>
                </body>
            </html>
        "
`;

exports[`basics does nothing if a link tag does not exist 1`] = `
"
            <!doctype html>
            <html>
                <head>
                </head>
                <body>
                </body>
            </html>
        "
`;
